bundles/binding/org.openhab.binding.sallegra/src/main/java/org/openhab/binding/sallegra/internal/SallegraBinding.java

```diff
@@ -0,0 +1,345 @@
+		return "Sallegra Refresh Service";
+	}
+	
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void execute() {
+        for (SallegraBindingProvider provider : providers) {
+            Collection<String> names = provider.getItemNames();
+            for (String name : names) {
+                logger.debug("Name {}", name);
+                SallegraBindingConfig bindingConfig = provider
+                        .getBindingConfigFor(name);
+                String deviceId = bindingConfig.getModuleName();
+
+                /*
+                 * check if a device with this id is already configured
+                 */
+                if (sallegraNodes.containsKey(deviceId)) {
+                    SallegraNode node = sallegraNodes.get(deviceId);
+
+                    /*
+                     * yes, there is a device with this id, now update it
+                     */
+                    String value = null;
+                    switch (bindingConfig.getCmdId()) {
+	                    case RELAY:
+	                        value = node.getRelay(bindingConfig.getCmdValue());
+	                        break;
+	                    case DIMMER:
+	                        value = node.getDimmer(bindingConfig.getCmdValue());
+	                        break;
+	                    case INPUT:
+	                        value = node.getInput(bindingConfig.getCmdValue());
+	                        break; 
+	                    default:
+	                        break;
+                    }
+                    //If a value is got returned update it in openhab
+                    if (value != null) {
+                        postUpdate(provider, bindingConfig.getItem(), value);
+                    }
+                } else {
+                    logger.error("Unknown deviceId \"{}\"", deviceId);
+                }
+            }
+        }
+		
+	}
+
+	
+	
+    private void postUpdate(SallegraBindingProvider provider, Item item,
+            final String value) {
+        Class<? extends Item> itemType = provider.getItemType(item.getName());
+        State state = createState(itemType, value);
+
+        if (state != null) {
+            if (item instanceof GenericItem) {
+                ((GenericItem) item).setState(state);
+            }
+            eventPublisher.postUpdate(item.getName(), state);
+        }
+    }   
+
+    private State createState(Class<? extends Item> itemType,
+            String transformedResponse) {
+        try {
+            if (itemType.isAssignableFrom(NumberItem.class)) {
+                return DecimalType.valueOf(transformedResponse);
+            } else if (itemType.isAssignableFrom(SwitchItem.class)) {
+                return OnOffType.valueOf(transformedResponse);
+            } else if (itemType.isAssignableFrom(DimmerItem.class)) {
+                return PercentType.valueOf(transformedResponse);
+            } else {
+                return StringType.valueOf(transformedResponse);
+            }
```

Comment:
It would be better to query an `Item` instance (when you process the binding configuration) with `getAcceptedDataTypes()` to get the list of accepted types, instead of hardcoding a mapping of items to types in the binding.